components/feature/toolbar/src/main/java/mozilla/components/feature/toolbar/ToolbarPresenter.kt

```diff
@@ -18,77 +23,59 @@ import mozilla.components.feature.toolbar.internal.URLRenderer
 @Suppress("TooManyFunctions")
 class ToolbarPresenter(
     private val toolbar: Toolbar,
-    private val sessionManager: SessionManager,
-    private val sessionId: String? = null,
+    private val store: BrowserStore,
+    private val customTabId: String? = null,
     urlRenderConfiguration: ToolbarFeature.UrlRenderConfiguration? = null
-) : SelectionAwareSessionObserver(sessionManager) {
+) {
 
     @VisibleForTesting
     internal var renderer = URLRenderer(toolbar, urlRenderConfiguration)
 
+    private var subscription: Store.Subscription<BrowserState, BrowserAction>? = null
+
     /**
      * Start presenter: Display data in toolbar.
      */
     fun start() {
```

Comment:
Should this implement `LifecycleAwareFeature` if `start` and `stop` are used?

Reply:
The actual feature does and delegates to its sub components like the presenter and interactor.